release/staging-resources-cdk/lib/ArtifactsBucketStack.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {CfnOutput, RemovalPolicy, Stack, StackProps} from 'aws-cdk-lib';
+import {Construct} from 'constructs';
+import {BlockPublicAccess, Bucket} from 'aws-cdk-lib/aws-s3';
+
+/**
+ * This stack creates an S3 bucket for artifacts. It is distinct from
+ * {@link StagingResourcesStack} to support using an existing bucket.
+ */
+export class ArtifactsBucketStack extends Stack {
+  constructor(scope: Construct, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const archivesBucketName: string = scope.node.tryGetContext('archivesBucketName');
+
+    const archivesBucket = new Bucket(this, 'StagingArtifactsBucket', {
```

Comment:
We are using `artifacts` and `archives` interchangeably. Let's align on one name.

Reply:
I'll use "archives" here. I'm considering "artifacts" to be anything Data Prepper creates, including Docker images.